tests/unit/common/vector/test_vector.py

```diff
@@ -1079,12 +1138,16 @@ def _dtype_to_cypher_type(dtype: T_DTYPE_LITERAL) -> str:
     }[dtype]
 
 
-def _vec_element_cypher_repr(value: t.Any) -> str:
-    if isinstance(value, float):
+def _vec_element_cypher_repr(value: t.Any, dtype: T_DTYPE_LITERAL) -> str:
+    if isinstance(value, float) and dtype in {"f32", "f64"}:
         if math.isnan(value):
             return "NaN"
         if math.isinf(value):
             return "Infinity" if value > 0 else "-Infinity"
+    if dtype == "f32":
+        # account for float32 precision loss
+        compressed = struct.unpack(">f", struct.pack(">f", value))[0]
+        return repr(compressed)
```

Comment:
is this that thing from the call the other day?

Reply:
Similar. If you take a very small f64 value and cast it to f32, it might end up being exactly 0.0 because f32s have fewer decimal (actually binary) places.